tasks.py

```diff
@@ -231,6 +233,91 @@ def dev_env(ctx, architecture="amd64", name="kind", cni=None):
         tmp.flush()
         run("kubectl apply -f {}".format(tmp.name), echo=True)
 
+    if protocol == "bgp":
+        print("Configuring MetalLB with a BGP test environment")
+        bgp_dev_env()
+    else:
+        print("Leaving MetalLB unconfigured")
+
+
+# Configure MetalLB in the dev-env for BGP testing.  Start an frr based BGP
+# router in a container and configure MetalLB to peer with it.
+# See dev-env/bgp/README.md for some more information.
+def bgp_dev_env():
+    dev_env_dir = os.getcwd() + "/dev-env/bgp"
+    frr_volume_dir = dev_env_dir + "/frr-volume"
+
+    # TODO -- The IP address handling will need updates to add support for IPv6
+
+    # We need the IPs for each Node in the cluster to place them in the BGP
+    # router configuration file (bgpd.conf).  Each Node will peer with this
+    # router.
+    node_ips = run("kubectl get nodes -o jsonpath='{.items[*].status.addresses"
+            "[?(@.type==\"InternalIP\")].address}'", echo=True)
```

Comment:
I'm not sure about this suggestion, the code I'm adding is horrible but without the ending new line the next command output ends up in the same line.

```suggestion
            "[?(@.type==\"InternalIP\")].address}{\"\\n\"}'", echo=True)
```

What do you think? Take it or leave it. I wanted that until I wrote it and saw how ugly it was :joy:

Reply:
sure, happy to make the output a little prettier.  This command is already pretty ugly!